src/uct/ib/base/ib_device.c

```diff
@@ -629,23 +641,49 @@ size_t uct_ib_mtu_value(enum ibv_mtu mtu)
     ucs_fatal("Invalid MTU value (%d)", mtu);
 }
 
-uint8_t uct_ib_to_fabric_time(double time)
+uint8_t uct_ib_to_qp_fabric_time(double time)
 {
     double to;
-    long t;
 
     to = log(time / 4.096e-6) / log(2.0);
     if (to < 1) {
         return 1; /* Very small timeout */
-    } else if (to > 30) {
+    } else if ((long)(to + 0.5) >= uct_ib_fabric_time_max) {
         return 0; /* No timeout */
     } else {
-        t = (long)(to + 0.5);
-        ucs_assert(t >= 1 && t < 31);
-        return t;
+        return (long)(to + 0.5);
```

Comment:
is 31 possible?

Reply:
yes, 31 is possible and will be handled in 654 line